src/components/elements/table/GenericTable.tsx

```
@@ -80,6 +80,47 @@ const clickableRowStyles = {
   cursor: 'pointer',
 };
 
+export const getStickyCellStyles = (
+  sticky?: 'left' | 'right' | undefined
+): SxProps<Theme> => {
+  if (!sticky) return {} as SxProps<Theme>;
+
+  const base = {
+    backgroundColor: 'background.paper', // Otherwise it's transparent and other cell content appears beneath it
+    position: 'sticky',
+  };
+
+  // Pseudo-element to achieve a border on sticky cells. `position: sticky` doesn't work with regular border
+  const pseudo = {
+    content: '""',
+    position: 'absolute',
+    top: 0,
+    bottom: 0,
+    width: '1px',
+    backgroundColor: 'borders.light',
+    pointerEvents: 'none', // Don't interfere with interactions
+  };
+
+  if (sticky === 'right')
+    return {
+      ...base,
+      right: 0,
+      '&::before': {
+        ...pseudo,
+        left: 0,
+      },
+    } as SxProps<Theme>;
```

TODO: typescript issue here

---

I fiddled around with this a bit, and think I figured out where the issue is coming from. The `SxProp<Theme>` can be an object OR an array, so you can't spread it directly. This is explained in the MUI docs here: https://mui.com/system/getting-started/the-sx-prop/#passing-the-sx-prop

Here is what I found:

* the casts inside the `getStickyCellStyles` function are unnecessary and can be removed. Using the return type is sufficient
* change the return type of `getStickyCellStyles` to be `SystemStyleObject<Theme>` so that we know it's an object, not an array
* instead of casting to `as SxProps<Theme>` when passing the combined sx, pull it out into a variable so that it can be safely type-schecked. This way we won't swallow type errors. Example below that passes type checking and is safe

L282:
```
          {columns.map((def, i) => {
            const headerStyles = headerCellSx ? headerCellSx(def) : {};
            const sx: SxProps<Theme> = {
              ...getStickyCellStyles(def.sticky),
              textAlign: def.textAlign,
              width: def.width,
              // headerStyles has SxProps type so we can't spread it directly. https://mui.com/system/getting-started/the-sx-prop/#passing-the-sx-prop
              ...(Array.isArray(headerStyles) ? headerStyles : [headerStyles]),
            };

            return (
              <HeaderCell key={key(def) || i} sx={sx}>
                {def.header ? (
                  <strong>{def.header}</strong>
                ) : (
                  // If header isn't provided, add a visually hidden header with the column key for accessibility
                  <Box sx={visuallyHidden}>{def.key}</Box>
                )}
              </HeaderCell>
            );
          })}
```

L460:
```
                      const sx: SxProps<Theme> = {
                        ...getStickyCellStyles(def.sticky),
                        width,
                        minWidth,
                        ...(isLinked ? { p: 0 } : undefined),
                        textAlign,
                        whiteSpace: 'initial',
                        ...onClickLinkTreatment,
                        ...tableCellProps?.sx,
                      };
                      return (
                        <TableCell
                          key={key(def) || index}
                          {...tableCellProps}
                          sx={sx}
                        >
```

---

Thank you!

---

> instead of casting to as SxProps<Theme> when passing the combined sx, pull it out into a variable so that it can be safely type-schecked. This way we won't swallow type errors. Example below that passes type checking and is safe

I haven't totally understood this part. Typescript seems happy without any casting without pulling it out into a variable, (see https://github.com/greenriver/hmis-frontend/pull/1023/commits/7f50da6300573fd7fd39a5aca6feccf6fdc74a71), but are you indicating that this still carries the risk of swallowing an error, even without the cast?

---

Ah ok, great! It looks like it's happy without it, so it shouldn't be necessary. That must have  just been something I tried that ultimately wasn't necessary.